src/otg/dataset/study_locus.py

```diff
@@ -1564,12 +1571,20 @@ def annotate_ld(
                     f.col("r2"),
                 ),
             )
+            .filter(f.col("r2Overall") >= min_r2)
             .groupBy("chromosome", "studyId", "variantId")
             .agg(
                 f.collect_set(
                     f.when(
                         f.col("tagVariantId").isNotNull(),
-                        f.struct("tagVariantId", "r2Overall"),
+                        f.struct(
+                            [
+                                f.lit(None).cast(datatype).alias(field)
```

Comment:
I don't think this is a good approach, but it's better not to block this PR considering the upcoming changes